lightning/src/chain/chainmonitor.rs

```diff
@@ -262,6 +345,54 @@ where C::Target: chain::Filter,
 		self.monitors.write().unwrap().remove(funding_txo).unwrap().monitor
 	}
 
+	/// Indicates the persistence of a [`ChannelMonitor`] has completed after
+	/// [`ChannelMonitorUpdateErr::TemporaryFailure`] was returned from an update operation.
+	///
+	/// Thus, the anticipated use is, at a high level:
+	///  1) This [`ChainMonitor`] calls [`Persist::update_persisted_channel`] which stores the
+	///     update to disk and begins updating any remote (e.g. watchtower/backup) copies,
+	///     returning [`ChannelMonitorUpdateErr::TemporaryFailure`],
+	///  2) once all remote copies are updated, you call this function with the update_id that
+	///     completed, and once all pending updates have completed the Channel will be re-enabled.
+	pub fn channel_monitor_updated(&self, funding_txo: OutPoint, completed_update_id: MonitorUpdateId) {
+		let monitors = self.monitors.read().unwrap();
+		let monitor_data = if let Some(mon) = monitors.get(&funding_txo) { mon } else { return; };
+		let mut pending_monitor_updates = monitor_data.pending_monitor_updates.lock().unwrap();
+		pending_monitor_updates.retain(|update_id| *update_id != completed_update_id);
+
+		match completed_update_id {
+			MonitorUpdateId { contents: MonitorUpdate::MonitorUpdateId(_) } => {
+				let monitor_update_pending_updates =  pending_monitor_updates.iter().filter(|update_id|
+					if let MonitorUpdate::MonitorUpdateId(_) = update_id.contents { true } else { false }).count();
+				if monitor_update_pending_updates != 0 {
+					// If there are still monitor updates pending, we cannot yet construct an
+					// UpdateCompleted event.
+					return;
+				}
+				self.user_provided_events.lock().unwrap().push(MonitorEvent::UpdateCompleted {
```

Comment:
Possible to document why it's OK if `SyncPersist` monitor updates are still pending when we push this event?